From 27d632208163a73a0501e595fcdef0302cb44d8c Mon Sep 17 00:00:00 2001
From: eaglegai <eaglegai@163.com>
Date: Tue, 1 Jun 2021 17:21:55 +0800
Subject: [PATCH] proftpd 1.3.7a Adjusting unit test timeouts for netacl

---
 tests/api/netacl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/api/netacl.c b/tests/api/netacl.c
index c4da486..86b628d 100644
--- a/tests/api/netacl.c
+++ b/tests/api/netacl.c
@@ -894,6 +894,8 @@ Suite *tests_get_netacl_suite(void) {
   tcase_add_test(testcase, netacl_match_test);
   tcase_add_test(testcase, netacl_get_negated_test);
 
+  tcase_set_timeout(testcase, 60);
+
   suite_add_tcase(suite, testcase);
   return suite;
 }
-- 
1.8.3.1

